Hints.py

```diff
@@ -750,9 +755,34 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
 
         if world.shopsanity != "off" and "Progressive Wallet" not in world.item_hints:
             world.item_hints.append("Progressive Wallet")
-        world.named_item_pool = list(world.item_hints)
-
+                
+
+    #Removes items from item_hints list if they are included in starting gear.
+    #This method ensures that the right number of copies are removed, e.g.
+    #if you start with one strength and hints call for two, you still get
+    #one hint for strength
+    for item in [*world.starting_items, *world.starting_equipment, *world.starting_songs]:
+        itemname = everything[item].itemname
+        if itemname in world.item_hints:
+            world.item_hints.remove(itemname)
+
+    #Skip_child_zelda can cause the same problem, but needs to be handled separately since
+    #it doesn't update the starting gear lists
+    if world.skip_child_zelda:
+        itemname = str(world.get_location('Song from Impa').item)
+        if itemname in world.item_hints:
+            world.item_hints.remove(itemname)
+
+    world.named_item_pool = list(world.item_hints)
+
+    #Make sure the total number of hints won't pass 40. If so, we limit the always and trial hints
+    if world.hint_dist == "bingo":
+        numTrialHints=[0,1,2,3,2,1,0]
```

Comment:
nit: add spaces to format this nicely.

Reply:
lololol I didn't even see 290f19e when I wrote this.

Reply:
I knew I was going to miss one; the original version I forgot to put any of them, and then remembered you mentioning it last PR LUL